components/org.wso2.carbon.data.provider/src/main/java/org/wso2/carbon/data/provider/endpoint/DataProviderEndPoint.java

```diff
@@ -103,6 +124,42 @@ public static void onOpen(WebSocketConnection webSocketConnection) {
     @OnMessage
     public void onMessage(String message, WebSocketConnection webSocketConnection) {
         DataProviderConfigRoot dataProviderConfigRoot = new Gson().fromJson(message, DataProviderConfigRoot.class);
+        String authoringClassName;
+        try {
+            Map webSocketConfiguration
+                    = (Map)getDataProviderHelper().getConfigProvider().getConfigurationObject(WEB_SOCKET_CONFIG_HEADER);
+            if (webSocketConfiguration != null) {
+                JSONObject webSocketConfigJSON = new JSONObject(new Gson().toJson(webSocketConfiguration));
+                if (!webSocketConfigJSON.has(WEB_SOCKET_AUTHORIZING_CLASS_CONFIG_HEADER)) {
+                    throw new Exception("Web socket authorizing class cannot be found in the deployment.yaml file.");
+                }
+                authoringClassName = (String) webSocketConfigJSON.get(WEB_SOCKET_AUTHORIZING_CLASS_CONFIG_HEADER);
+            } else {
+                authoringClassName = DEFAULT_WEBSOCKET_AUTHORIZING_CLASS;
+            }
+            DataProviderAuthorizer dataProviderAuthorizer;
+            try {
+                dataProviderAuthorizer = getDataProviderHelper().getDataProviderAuthorizer(authoringClassName);
+            } catch (NullPointerException e) {
+                throw new Exception("Cannot find the Data Provider Authorizer class for the given class name: "
+                        + authoringClassName + ".");
+            }
+            boolean authorizerResult = dataProviderAuthorizer.authorize(dataProviderConfigRoot);
+            if (!authorizerResult) {
+                throw new Exception("Access denied to data provider.");
+            }
+        } catch (Exception e) {
```

Comment:
What is the need for catching general exception?

Reply:
In here three exceptions are thrown[1,2,3]. Once an exception is thrown we need to catch that because as this is related to a WebSocket connection, we need to send a text to WebSocket client about the error. In order to do that we catch the general exception and sends the text. Also please note that in [1,2,3] scenarios, we throw a general exception as by design this class did not have a specific exception implemented.

[1] https://github.com/wso2/carbon-analytics/blob/master/components/org.wso2.carbon.data.provider/src/main/java/org/wso2/carbon/data/provider/endpoint/DataProviderEndPoint.java#L139
[2] https://github.com/wso2/carbon-analytics/blob/master/components/org.wso2.carbon.data.provider/src/main/java/org/wso2/carbon/data/provider/endpoint/DataProviderEndPoint.java#L149
[3] https://github.com/wso2/carbon-analytics/blob/master/components/org.wso2.carbon.data.provider/src/main/java/org/wso2/carbon/data/provider/endpoint/DataProviderEndPoint.java#L154